x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/vector/Knn.java

```diff
@@ -139,14 +144,23 @@ public Knn(
             optional = true
         ) Expression options
     ) {
-        this(source, field, query, k, options, null);
+        this(source, field, query, k, options, null, List.of());
     }
 
-    private Knn(Source source, Expression field, Expression query, Expression k, Expression options, QueryBuilder queryBuilder) {
+    private Knn(
+        Source source,
+        Expression field,
+        Expression query,
+        Expression k,
+        Expression options,
+        QueryBuilder queryBuilder,
+        List<Expression> filterExpressions
+    ) {
         super(source, query, expressionList(field, query, k, options), queryBuilder);
         this.field = field;
         this.k = k;
         this.options = options;
+        this.filterExpressions = filterExpressions;
```

Comment:
we need to add both `filterExpressions` and `queryBuilder` to the `nodeInfo` method:

https://github.com/elastic/elasticsearch/blob/0b06d9f24dbac94f89bf696daad98041b93e7b1b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/vector/Knn.java#L292-294

my understanding is that if we don't add them, we risk losing both `filterExpressions` and `queryBuilder` during transforms.

Reply:
Thanks for the catch - added in b347bbacf702d94e2c53b4eabeadaadc08159a65